packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -109,11 +102,15 @@ class _ListenerEntry extends LinkedListEntry<_ListenerEntry> {
 ///
 ///  * [ValueNotifier], which is a [ChangeNotifier] that wraps a single value.
 class ChangeNotifier implements Listenable {
-  LinkedList<_ListenerEntry>? _listeners = LinkedList<_ListenerEntry>();
+  int _length = 0;
+  List<VoidCallback?> _listeners = List<VoidCallback?>.filled(0, null);
+  int _notificationCallStackDepth = 0;
+  int _removedListeners = 0;
```

Comment:
for clarity, it might be good to rename this `_reentrantlyRemovedListeners` or some such, since it doesn't track in general the number of removed listeners, it only tracks it at certain times.